### DISCUSSION TITLE
Run short command without autocompletion

### DISCUSSION THREAD

**Discussion by illukin** [daniele77/cli#186] [Ideas / Feature requests]
Hi!

I have a few commands, for example:
```
Root menu:
    - help
    - show
```

If I type `h` and press `TAB` I will get `help` and after that I can press `ENTER` and execute help command.
But I want just type `h` and after that press `ENTER` without press `TAB` button.

Is it possible to run short command if we can explicitly identify it?

It is useful when you should type long command, for example:
Instead of `show interface eth0 status` you can simply type `s i eth0 s`

**Comment by daniele77** [comment]
This is currently not feasible, and that would require new development.
Actually, I don't plan to introduce this feature soon.
However, you can manually add the commands "s", "I" recalling to the same handlers of "show" and "interface".

**Comment by illukin** [comment]
> However, you can manually add the commands "s", "I" recalling to the same handlers of "show" and "interface".

Of course I can )
But the main idea of this feature is that you can type any unique sequence of symbols to explicitly identify a command (not only first symbol).

So you can type `s` or `sh` or `sho` or `show` and run `show` command.

> This is currently not feasible, and that would require new development.

I understand.

**Reply by daniele77** [reply]
What I can do is check whether adding such a feature is a big effort and if it is worth it (in that case I can plan the development and open a GitHub issue.
Of course, PR are welcome :-)

**Reply by daniele77** [reply]
I checked the code. Unfortunately, the architecture does not support short commands.
It would be pretty challenging to add this feature because the check `input == <cmd name>` is performed inside the command object: unfortunately, the command has not the visibility of the other commands inside the menu, to check whether the other commands don't start with the same prefix.
